.travis.yml

```diff
@@ -56,30 +56,22 @@ jobs:
     after_success: skip
   - stage: docs
     python: '3.6'
-    addons:
-      chrome: stable
-      apt:
-        packages:
-          - chromium-chromedriver
     before_install:
       - sudo apt-get update
       - sudo apt-get -qq install pandoc
       - sudo apt-get -qq install libtcmalloc-minimal4
       - export LD_PRELOAD="/usr/lib/libtcmalloc_minimal.so.4"
       - pip install --upgrade pip setuptools wheel
-      - pip install selenium
     install:
       - pip install --ignore-installed -U -q -e .[complete]
       - pip freeze
-    before_script:
-      - ln --symbolic /usr/lib/chromium-browser/chromedriver "${HOME}/bin/chromedriver"
     script:
       - python -m doctest README.md
       - cd docs && make html && cd -
       - touch docs/_build/html/.nojekyll
-      # Open Binder example in browser to trigger repo2docker to build image
-      - python binder/trigger_binder.py --url https://mybinder.org/v2/gh/diana-hep/pyhf/master
-      - python binder/trigger_binder.py --url https://mybinder.org/v2/gh/diana-hep/pyhf/master?filepath=docs%2Fexamples%2Fnotebooks%2Fbinderexample%2FStatisticalAnalysis.ipynb
+      # Use Binder build API to trigger repo2docker to build image
+      - python -m webbrowser "https://mybinder.org/build/gh/diana-hep/pyhf/master"
+      - python -m webbrowser "https://mybinder.org/build/gh/diana-hep/pyhf/master?filepath=docs%2Fexamples%2Fnotebooks%2Fbinderexample%2FStatisticalAnalysis.ipynb"
```

Comment:
FWIW, filepath isn't part of the build, it is only consumed by the browser redirects after build completes. These two lines won't do different things.

Additionally, you might want to use travis [environment variables](https://docs.travis-ci.com/user/environment-variables/#default-environment-variables) to ensure that the PR is being tested. I *think* that `https://mybinder.org/build/gh/$TRAVIS_REPO_SLUG/$TRAVIS_COMMIT` will be right. You *might* need to use $TRAVIS_PULL_REQUEST_SLUG for PRs since I believe that PR commits are available as refs in the target repo even before it's merged. Note that binder builds are cached by commit sha (it resolves branches and tags to a commit before checking), so building `abc123` and then requesting `master` later will use the cached build as long as the commits match.

Alternately, if this isn't really a test and you only want to build master, moving it to the `deploy` stage might be appropriate, since that only runs on master, not PRs.

Reply:
@minrk thanks for the feedback!

>  filepath isn't part of the build, it is only consumed by the browser redirects after build completes. These two lines won't do different things.

Ah, so if I understand correctly, you're saying that I have been misunderstanding how Binder works and if I have Binder build an image off of the `master` branch then that image will be used for all subsequent requests off of that branch _regardless_ of filepaths. This makes sense now that I think about it, but I think I must have had some confusion in the past that if I requested a URL with `filepath=` in it then this actually built a *different* image then if `filepath=` wasn't there.

> Alternately, if this isn't really a test and you only want to build master, moving it to the `deploy` stage might be appropriate, since that only runs on master, not PRs.

This is a good point, as this is really what we're doing.

Reply:
> if I have Binder build an image off of the master branch then that image will be used for all subsequent requests off of that branch regardless of filepaths

Correct. Binder has two stages: build and launch. The inputs to the build are only the repo and ref, while the 'filepath' is an input to launch, indicating what URL to redirect users to when the server is ready.